packages/wallets/src/okx/okx.page.ts

```diff
@@ -1,201 +1,268 @@
-      await this.page.waitForTimeout(2000);
-      await this.page.getByRole('button', { name: 'Confirm' }).click();
-
-      // Dive into wallet main page after installation
-      await this.page.click("button:has-text('Start your Web3 journey')");
-      // Wait until extension to be loaded after installation with ETH value display.
-      // ETH value displayed with ETH symbol
-      await this.page.waitForSelector('text=ETH', { state: 'visible' });
-      //Looks like after installation and load extension mainPage there we should to wait a bit for extension make sure to be installed in some memory
-      await this.page.waitForTimeout(2000);
+  /** Get token balance from wallet extension using `tokenName` */
+  async getTokenBalance(tokenName: string) {
+    return await test.step(`Get ${tokenName} token balance`, async () => {
+      await this.navigate();
+      const amount = parseFloat(await this.homePage.getTokenBalance(tokenName));
+      await this.page.close();
+      return amount;
     });
   }
 
+  /** Add new network to wallet */
   async addNetwork(
     networkName: string,
     networkUrl: string,
     chainId: number,
     tokenSymbol: string,
+    blockExplorer?: string,
   ) {
-    await test.step('Add network', async () => {
-      if (!this.page) throw "Page isn't ready";
-      await this.navigate();
-      await this.page.click('data-testid=account-options-menu-button');
-      await this.page.click('text=Settings');
-      await this.page.click("text='Networks'");
-      await this.page.click('text=Add a network');
-      await this.page.click("a :has-text('Add a network manually')");
-      await this.page.fill(
-        ".form-field :has-text('Network Name') >> input",
+    if (!(await isNeedAddNetwork(networkName))) {
+      return;
+    }
+    await this.navigate();
+    await this.homePage.networkListButton.click();
+    if (
+      !(await this.networkListPage.isNetworkExist(
+        await checkNetworkName(networkName),
+      ))
+    ) {
+      await this.networkListPage.addCustomNetwork(
         networkName,
-      );
-      await this.page.fill(
-        ".form-field :has-text('New RPC URL') >> input",
         networkUrl,
-      );
-      await this.page.fill(
-        ".form-field :has-text('Chain ID') >> input",
-        String(chainId),
-      );
-      await this.page.fill(
-        ".form-field :has-text('Currency symbol') >> input",
+        chainId,
         tokenSymbol,
+        blockExplorer,
       );
-      await this.page.click('text=Save');
+    }
+    await this.page.close();
+  }
+
+  /** Switch network in the wallet
+   * - switch in wallet extension
+   * - switch in the connected dApp (switch in the extension doesn't switch the dApp network, but we do it to sync states)
+   * */
+  async changeNetwork(networkName: string) {
+    await test.step(`Switch network to "${networkName}"`, async () => {
+      networkName = await checkNetworkName(networkName);
```

Comment:
why checkNetworkName return something? Could you rename it for `convertNetworkName` or `getCorrectNetworkName`?